### PR TITLE
Put some guards in place to make sure CFG builder doesn’t violate IOperation invariants while visiting malformed IOperation tree, which didn’t violate those invariants.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@333fred, @mavasani, @cston, @dotnet/roslyn-compiler, @dotnet/analyzer-ioperation Please review

**Review by mavasani — APPROVED** [review]

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs
Comment (333fred) [inline src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: The fact that we only need the result parameter in 2 cases feels like that means that we should just call `AddStatement` there, and return null in all cases. We're asserting in `StartVisitingStatement` that the `if` below should always be taken, so I don't see any big problems with this, and it will make our implementation more standard.

**Review by 333fred — APPROVED** [review]
Done review pass (iteration 1). If my concern is the only one raised during review, I think it's fine to address with a follow up if necessary.

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler Please review, need second review

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs
Comment (cston) [inline src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: >originalOperaion [](start = 62, length = 16)

Typo: `Operation`

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs
Comment (cston) [inline src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: Where is this overload used?

**Review by cston — APPROVED** [review]

**Pull request merged**
